### ISSUE TITLE
Expose custom Logs module(s) and/or source

### ISSUE THREAD

**Issue by hcarty** [mirage/ocaml-cohttp#645]
The move to using Logs for logging is wonderful!  However, it makes it almost impossible to use debug-level logging in an application using cohttp as cohttp does not expose its custom logging source.  That means that an application passing around large blob of data with cohttp ends up dumping all of those bytes as log messages when an application's log verbosity is set to `debug`.

It would be useful to expose enough of the internal logging configuration to allow an application to explicitly set cohttp's log level separately from the rest of the application.

**Comment by osener** [comment]
Is there a way to silence Cohttp's log messages or redirect them somewhere else? I use Logs library in my application as well, and Cohttp's log messages make it very hard to follow what's going on during development in a terminal emulator.

**Comment by mseri** [comment]
If you see cohttp logging you probably set the `debug` level for all sources.
In this case, you can mute all cohttp logging as follows (remember to run this each time you set the log level with `~all:true` and don't use those as your sources names).

```ocaml
List.iter (fun src ->
      let name = Logs.Src.name src in
      if List.mem name [ "cohttp.lwt.io"; "cohttp.lwt.server" ] then
        Logs.Src.set_level src None)
  @@ Logs.Src.list ();
```

**Comment by mseri** [comment]
I think we should document this better, and provide those labels somewhere, but I have experimented with it for a bit and I wouldn't really know how else to change the API.

**Comment by mseri** [comment]
I am going to close this issue, if there are further ideas for improvements feel free to reopen it or start a discussion in https://github.com/mirage/ocaml-cohttp/discussions
